### PR TITLE
[async] Add advection benchmark

### PR DESCRIPTION
**Pull request by yuanming-hu**
Related issue = #

 
 -  Added `advection_2d`
 - Reduced data size of async benchmark cases by 10x since now we have IRHandles
 - Added `device_time` recording

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

**Review by k-ye — APPROVED** [review]
LGTM!

File: benchmarks/async_advection.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        gui = ti.GUI('Advection schemes', (n, n))
+        for i in range(10):
+            for _ in range(10):
+                advect()
+            gui.set_image(x.to_numpy())
+            gui.show()
+
+
+if __name__ == '__main__':
+    advection_2d()
```
Comment (xumingkuan) [inline benchmarks/async_advection.py:125]: I wonder if the `scale` parameter is missing here?

**Review by xumingkuan — APPROVED** [review]
LGTM with a nit!

File: benchmarks/async_advection.py (hunk 1 above)
Comment (yuanming-hu) [reply benchmarks/async_advection.py:125]: I know it's confusing, but the `@benchmark_async` decorator takes care of that `scale` parameter :-)

**Pull request merged**